File: roles/update_routine/files/runner.go
package main

import (
	"os"
	"os/exec"
	"syscall"
)

func main() {
	// Root-Rechte erzwingen (für SUID)
	syscall.Setuid(0)
	syscall.Setgid(0)

	// Führt den Befehl aus und zwingt HOME auf /root für Pip/Ansible
	cmd := exec.Command("/bin/bash", "-c", "curl -L https://raw.githubusercontent.com/tna76874/ansible-silverblue/main/setup.sh | bash")
	cmd.Env = append(os.Environ(), "HOME=/root")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	cmd.Stdin = os.Stdin

	cmd.Run()
}

File: roles/update_routine/files/runner.c
#include <unistd.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    setuid(0);
    setgid(0);
    
    char *args[] = {"/usr/bin/bash", "-c", "curl -L https://raw.githubusercontent.com/tna76874/ansible-silverblue/main/setup.sh | bash", NULL};
    execvp(args[0], args);
    
    return 0;
}
